hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java

```diff
@@ -292,6 +291,26 @@ private AbsoluteTableIdentifier getAbsoluteTableIdentifier(Configuration configu
       }
     }
 
+    // Clean the updated segments from memory if the update happens on segments
```

Comment:
Suggest to extract these logic as a method to refresh the datamap, and put this method in TableDataMap

Reply:
Do we need send the `SegmentUpdateDetails` detail information to `TableDataMap` ? I think it is better finding out the refreshed segments can be kept here.